tests/functional/listing/test.js

```diff
@@ -0,0 +1,476 @@
+'use strict';
+const AWS = require('aws-sdk');
+const async = require('async');
+const assert = require('assert');
+
+function useAws() {
+    return !!process.env.REAL_AWS;
+}
+
+function cutAttributes(data) {
+    const newContent = [];
+    const newPrefixes = [];
+    data.Contents.forEach(item => {
+        newContent.push(item.Key);
+    });
+    data.Contents = newContent;
+    data.CommonPrefixes.forEach(item => {
+        newPrefixes.push(item.Prefix);
+    });
+    data.CommonPrefixes = newPrefixes;
+    if (data.NextMarker === '') {
+        delete data.NextMarker;
+    }
+    if (data.EncodingType === '') {
+        delete data.EncodingType;
+    }
+    if (data.Delimiter === '') {
+        delete data.Delimiter;
+    }
+}
+
+const Bucket = 'listing-test-ggim';
+
+const objects = [
+    { Bucket, Key: 'Pâtisserie=中文-español-English', Body: '' },
+    { Bucket, Key: 'notes/spring/1.txt', Body: '' },
+    { Bucket, Key: 'notes/spring/2.txt', Body: '' },
+    { Bucket, Key: 'notes/spring/march/1.txt', Body: '' },
+    { Bucket, Key: 'notes/summer/1.txt', Body: '' },
+    { Bucket, Key: 'notes/summer/2.txt', Body: '' },
+    { Bucket, Key: 'notes/summer/august/1.txt', Body: '' },
+    { Bucket, Key: 'notes/year.txt', Body: '' },
+    { Bucket, Key: 'notes/yore.rs', Body: '' },
+    { Bucket, Key: 'notes/zaphod/Beeblebrox.txt', Body: '' },
+];
+
+if (useAws()) {
```

Comment:
for the setup, let's do what we are doing for the other s3 aws tests.  see this file for example: https://github.com/scality/S3/blob/master/tests/functional/aws-node-sdk/test/bucket/testBucketStress.js#L4